pinecone/data/vector_factory.py

```diff
@@ -91,7 +91,7 @@ def _dict_to_vector(item, check_type: bool) -> Vector:
         try:
             return Vector(**item, _check_type=check_type)
         except TypeError as e:
-            if not isinstance(item["values"], Iterable) or not isinstance(item["values"][0], numbers.Real):
+            if not isinstance(item["values"], Iterable) or not isinstance(item["values"].__iter__().__next__(), numbers.Real):
```

Comment:
Isn't item["values"][0] better in terms of readability? What's the reason for this change?

Reply:
It was better readability, but it was a type error because in python something that is `Iterable` is not necessarily subscriptable with `[0]`. 

To be `Iterable` something must implement the `__iter__()` method and returns an iterator (i.e. something implementing `__next__()`). In normal usage, these methods are not referenced directly... much more common to see iterators used in list comprehensions or in for-loop statements.

To use indexes like `foo[0]`, a class must implement the magic method `__getitem__()`

So this was a type error because not all iterables support indexing like `foo[0]`. Even though python lists (basically arrays) are both iterable and indexable, that's not always the case for all `Iterable` items from the compiler's perspective.

Relevant docs: 
- https://wiki.python.org/moin/Iterator
- https://docs.python.org/3/reference/datamodel.html#object.__getitem__

Reply:
Thanks for the explanation. 👍